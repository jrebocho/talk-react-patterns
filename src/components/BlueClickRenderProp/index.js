import React, { Component } from 'react'
import './BlueClickRenderProp.css'

class BlueClickRenderProp extends Component {
  constructor(props) {
    super(props)
    this.state = { on: false }
    this.toggle = this.toggle.bind(this)
  }

  toggle() {
    this.setState(prevState => ({
      on: !prevState.on
    }))
  }

  render() {
    return (
      <div className="blue-box" onClick={this.toggle}>
        {this.props.render(this.state.on)}
      </div>
    )
  }
}

export default BlueClickRenderProp
