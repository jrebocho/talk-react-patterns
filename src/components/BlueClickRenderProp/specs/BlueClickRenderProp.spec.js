import React from 'react'
import { shallow } from 'enzyme'
import BlueClickRenderProp from '../../BlueClickRenderProp'

describe('<BlueClickRenderProp />', () => {
  it('renders correctly', () => {
    const TestComponent = () => <div>test</div>

    const wrapper = shallow(
      <BlueClickRenderProp render={on => <TestComponent on={on} />} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is clicked', () => {
    it('passes the on prop with true', () => {
      const TestComponent = () => <div>test</div>

      const wrapper = shallow(
        <BlueClickRenderProp render={on => <TestComponent on={on} />} />
      )
      wrapper.simulate('click')

      expect(wrapper.find('TestComponent').prop('on')).toBeTruthy()
    })
  })
})
