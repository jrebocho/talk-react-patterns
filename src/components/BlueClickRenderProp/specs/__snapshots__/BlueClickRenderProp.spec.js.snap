// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlueClickRenderProp /> renders correctly 1`] = `
<div
  className="blue-box"
  onClick={[Function]}
>
  <TestComponent
    on={false}
  />
</div>
`;
